data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+    regex prefix; to return the research_id for the specific pid passed
+
+    :param project: bq name of project
+    :param dataset: bq name of dataset
+    :param pid: person_id passed to receive associated research_id
+    :return: research_id or None if it does not exist for that person_id
+    """
+    # Get non deid dataset prefix
+    prefix = dataset.split('_')[0]
+    prefix = prefix.replace('R', '')
+
+    research_id_df = bq.query(
+        RESEARCH_ID_QUERY.format(project=project, prefix_regex=prefix, pid=pid))
+    if research_id_df.empty:
+        logging.info('no research_id associated with person_id: %s ' % pid)
+        return None
+    else:
+        return research_id_df.to_string()
+
+
+def check_pid_exist(pid, date_row):
+    """
+    Queries the table that retraction will take place, to see if the PID exists before creating query
+    :param pid: person_id
+    :param date_row:
+    :return:
+    """
+    check_pid_df = bq.query(
+        CHECK_PID_EXIST_QUERY.format(project=date_row.project_id,
+                                     dataset=date_row.dataset_id,
+                                     table=date_row.table,
+                                     pid=pid))
+    return bool(check_pid_df.get_value(0, 'count') > 0)
+
+
+def create_queries(project_id, ticket_number, pids_project_id, pids_dataset_id,
+                   pids_table):
+    """
+    Creates sandbox and truncate queries to run for EHR deactivated retraction
+
+    :param project_id: bq name of project
+    :param ticket_number: Jira ticket number to identify and title sandbox table
+    :param pids_project_id: deactivated ehr pids table in bq's project_id
+    :param pids_dataset_id: deactivated ehr pids table in bq's dataset_id
+    :param pids_table: deactivated pids table in bq's table name
+    :return: list of queries to run
+    """
+    queries_list = []
+    # Hit bq and receive df of deactivated ehr pids and deactivated date
+    deactivated_ehr_pids_df = bq.query(
+        DEACTIVATED_PIDS_QUERY.format(project=pids_project_id,
+                                      dataset=pids_dataset_id,
+                                      table=pids_table), project_id)
+
+    date_columns_df = get_date_info_for_pids_tables(project_id)
+    logging.info(
+        'Dataframe creation complete to loop through to create retraction queries'
+    )
+
+    # Loop through the deactivated PIDS df to create queries based on the retractions needed per PID table
+    logging.info(
+        'looping through deactivated pids df to create queries based on teh retractions needed per pid table'
+    )
+    for ehr_row in deactivated_ehr_pids_df.itertuples(index=False):
+        for date_row in date_columns_df.itertuples(index=False):
+            # Determine if dataset is deid to correctly pull pid or research_id and check if ID exists in dataset or if
+            # already retracted
+            if is_deid_dataset(date_row.dataset_id) is True:
```

Comment:
Is this helper function really needed?  This could be rewritten as:
```
if re.match(DEID_REGEX, date_row.dataset_id):
```
If a match is returned, it is automatically True.  If no match exists, it is automatically false.  What is your design rationale?  Are you planning to use this somewhere else?

Reply:
I like your way better as well. I initially thought I may need that helper function more than once, but now that I see I don't, it's unnecessary and clutters, thanks!